pkg/serializer/internal/metrics/iterable_series.go

```diff
@@ -500,17 +500,35 @@ func (pb *PayloadsBuilder) finishPayload() error {
 // FIXME(maxime): to be removed when v2 endpoints are available
 func (series *IterableSeries) MarshalJSON() ([]byte, error) {
 	// use an alias to avoid infinite recursion while serializing a Series
-	type SeriesAlias Series
+	type SeriesAlias struct {
```

Comment:
Now that this is no longer a type alias, would it make sense to rename the type? The comment above about recursion is probably not applicable too.

```suggestion
	type SeriesWithMetadata struct {
```